src/azure-cli-core/azure/cli/core/extension/_resolve.py

```diff
@@ -88,6 +88,13 @@ def resolve_from_index(extension_name, cur_version=None, index_url=None, target_
     download_url, digest = chosen.get('downloadUrl'), chosen.get('sha256Digest')
     if not download_url:
         raise NoExtensionCandidatesError("No download url found.")
+    ext_endpoint = cli_ctx.cloud.endpoints.extension_storage_account_resource_id if cli_ctx and \
+        cli_ctx.cloud.endpoints.has_endpoint_set('extension_storage_account_resource_id') else None
+    config_index_url = cli_ctx.config.get('extension', 'index_url', None) if cli_ctx else None
```

Comment:
Maybe we can put these 2 URLs in one place?

Reply:
Also, isn't the `whl` location hard-coded in [the index](https://raw.githubusercontent.com/Azure/azure-cli-extensions/master/src/index.json)?

```
{
    "downloadUrl": "https://azurecliprod.blob.core.windows.net/cli-extensions/account-0.2.0-py2.py3-none-any.whl",
    "filename": "account-0.2.0-py2.py3-none-any.whl",
    "metadata": {
        "azext.isExperimental": true,
```

Reply:
The index will be teleported into different storage accounts in air gapped clouds, we do not have permission to modify the whl `downloadUrl` based on the destination storage accounts after it gets copied in air gapped clouds. So we need to modify the `downloadUrl` here in azure-cli when we need to download it.

Reply:
Do we really need to put one URL in cloud config, one in `az config`?

Reply:
The one in cloud config is now a general support for downloading extensions and other tools/files needed in air gapped clouds. It's always a storage account endpoint and controlled by Microsoft.

The one in `az config` is to support customers that want to use their own customized `index.json`, which may have their own private extensions and some allowed public extensions. You can use any url of the index file, such as the [raw content url](https://raw.githubusercontent.com/Azure/azure-cli-extensions/master/src/index.json) on a github branch.